tests/forcing/test_marrmot.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from esmvalcore.experimental import Recipe
+from esmvalcore.experimental.recipe_output import OutputFile
+
+from ewatercycle.forcing import generate, load
+from ewatercycle.forcing.marrmot import MarrmotForcing
+
+
+def test_plot():
+    f = MarrmotForcing(
+        start_time='1989-01-02T00:00:00Z',
+        end_time='1999-01-02T00:00:00Z',
+    )
+    with pytest.raises(NotImplementedError):
+        f.plot()
+
+
+@pytest.fixture
+def mock_recipe_run(monkeypatch, tmp_path):
+    """Overload the `run` method on esmvalcore Recipe's."""
+    data = {}
+
+    class MockTaskOutput:
+        fake_forcing_path = str(tmp_path / 'marrmot.mat')
+        files = (
+            OutputFile(fake_forcing_path),
+        )
+
+    def mock_run(self):
+        """Store recipe for inspection and return dummy output."""
+        nonlocal data
+        data['data_during_run'] = self.data
+        return {'diagnostic_daily/script': MockTaskOutput()}
+
+    monkeypatch.setattr(Recipe, "run", mock_run)
+    return data
```

Comment:
Cool, I really like the way this stores the data inside the fixture. I'm just not sure I understand where it is that this `data` is returned to?

Reply:
On line 36 I return the data dict. Because `mock_recipe_run` is a fixture. When used in test as an argument the value will be equal to data. 

recipe.run()` is called in the test so I had to scope data to nonlocal so I could store the arguments of the call to `run()`.
I could not do `data = self.data` because then in the test, the value of the fixture would still point to the line 21

Reply:
I renamed the outer `data` var to `recorder` to make it more clear